### PR TITLE
fix(variance): Fix theme computations and add more tests for prop behavior

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Resolved an issue where function values would be skipped if not placed inside a media query,
  * Worked: `colors={[({ colors }) => colors.blue]}`
  * Broken: `colors={({ colors }) => colors.blue}`
* Tests behavior around extra transform arguments, including cases where multiple properties were controlled by the same prop conditionally.
* Added tests around cases where transformation should not happen (global values when a scale had been applied).
* Converted some brittle tests to snapshots as their behaviors are known to be accurate.  May move these to component render validation as they are not as useful as unit tests.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.9.2-alpha.b57b69.0<br/>@codecademy/gamut-illustrations@0.17.20-alpha.b57b69.0<br/>@codecademy/gamut-kit@0.5.37-alpha.b57b69.0<br/>@codecademy/gamut-labs@19.22.3-alpha.b57b69.0<br/>@codecademy/gamut-patterns@0.5.8-alpha.b57b69.0<br/>@codecademy/gamut-styles@11.9.2-alpha.b57b69.0<br/>@codecademy/gamut-system@0.7.3-alpha.b57b69.0<br/>@codecademy/gamut-tests@2.4.23-alpha.b57b69.0<br/>@codecademy/gamut@36.4.5-alpha.b57b69.0<br/>@codecademy/konamimojisplosion@0.2.54-alpha.b57b69.0<br/>@codecademy/markdown-overrides@0.5.66-alpha.b57b69.0<br/>@codecademy/styleguide@43.7.5-alpha.b57b69.0<br/>@codecademy/variance@0.17.4-alpha.b57b69.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://611d2a7ee7930d1131ad4f90--gamut-preview.netlify.app>https://611d2a7ee7930d1131ad4f90--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/611d2a7ee7930d1131ad4f90>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.9.2-alpha.2ed92f.0<br/>@codecademy/gamut-illustrations@0.17.20-alpha.2ed92f.0<br/>@codecademy/gamut-kit@0.5.37-alpha.2ed92f.0<br/>@codecademy/gamut-labs@19.22.3-alpha.2ed92f.0<br/>@codecademy/gamut-patterns@0.5.8-alpha.2ed92f.0<br/>@codecademy/gamut-styles@11.9.2-alpha.2ed92f.0<br/>@codecademy/gamut-system@0.7.3-alpha.2ed92f.0<br/>@codecademy/gamut-tests@2.4.23-alpha.2ed92f.0<br/>@codecademy/gamut@36.4.5-alpha.2ed92f.0<br/>@codecademy/konamimojisplosion@0.2.54-alpha.2ed92f.0<br/>@codecademy/markdown-overrides@0.5.66-alpha.2ed92f.0<br/>@codecademy/styleguide@43.7.5-alpha.2ed92f.0<br/>@codecademy/variance@0.17.4-alpha.2ed92f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://611d3d051ff49c116c0ab4a9--gamut-preview.netlify.app>https://611d3d051ff49c116c0ab4a9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/611d3d051ff49c116c0ab4a9>Deploy Logs</a></p>

File: packages/variance/integration/__tests__/variance-test.ts [hunk 1]
```diff
@@ -279,7 +258,7 @@ describe('css', () => {
   it('allows selectors with media queries', () => {
     const returnedFn = css({
       width: ['100%', '200%'],
-      boxShadow: [({ colors }) => `0px 0px 0px 0px ${colors.black}`],
+      boxShadow: ({ colors }) => `0px 0px 0px 0px ${colors.black}`,
```
Comment (JoshuaKGoldberg) [inline packages/variance/integration/__tests__/variance-test.ts:261]: 👌  I have a vague memory about seeing this before...

**Review by JoshuaKGoldberg — APPROVED** [review]
🎉

File: packages/variance/src/types/config.ts [hunk 2]
```diff
@@ -126,9 +130,10 @@ export type ParserProps<
 >;
 
 export type SystemProps<P extends AbstractParser> = {
-  [K in keyof Omit<Parameters<P>[0], 'theme'>]:
-    | Omit<Parameters<P>[0], 'theme'>[K]
-    | ResponsiveProp<(theme: Theme) => Omit<Parameters<P>[0], 'theme'>[K]>;
+  [K in keyof Omit<Parameters<P>[0], 'theme'>]: Omit<
+    Parameters<P>[0],
+    'theme'
+  >[K];
```
Comment (dreamwasp) [inline packages/variance/src/types/config.ts:136]: nice

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.9.3-alpha.651073.0<br/>@codecademy/gamut-illustrations@0.17.20-alpha.651073.0<br/>@codecademy/gamut-kit@0.5.51-alpha.651073.0<br/>@codecademy/gamut-labs@19.23.14-alpha.651073.0<br/>@codecademy/gamut-patterns@0.5.8-alpha.651073.0<br/>@codecademy/gamut-styles@11.9.2-alpha.651073.0<br/>@codecademy/gamut-tests@2.4.23-alpha.651073.0<br/>@codecademy/gamut@36.8.2-alpha.651073.0<br/>@codecademy/konamimojisplosion@0.2.54-alpha.651073.0<br/>@codecademy/markdown-overrides@0.5.67-alpha.651073.0<br/>@codecademy/monaco@0.3.5-alpha.651073.0<br/>@codecademy/styleguide@43.10.7-alpha.651073.0<br/>@codecademy/variance@0.17.4-alpha.651073.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://612d106f60675a00c333f3a7--gamut-preview.netlify.app>https://612d106f60675a00c333f3a7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/612d106f60675a00c333f3a7>Deploy Logs</a></p>

**Pull request merged**